database/scripts/lib/buildfarm_tools.rb

```diff
@@ -128,13 +131,49 @@ def self.jobs_last_success_date(older_than_days: 0)
     out
   end
 
-  def self.test_regressions_known
+  def self.test_regressions_known(sort_by: 'priority')
     out = known_issues(status: 'open')
     out.concat known_issues(status: 'disabled')
     out = out.group_by { |e| e["github_issue"] }.to_a.map { |e| e[1] }
+    out.each do |error_list|
+      priority = issue_priority(error_list.first["github_issue"])
+      error_list.each do |error|
+        error["priority"] = priority
+      end
+    end
+
+    unless sort_by.nil?
+      out.sort_by! { |r| -r.first['priority'] }
+    end
     out
   end
 
+  def self.issue_priority(issue_link)
+    sql_out = run_command('./sql_run.sh get_known_issue_errors.sql', args: [issue_link])
+    errors = sql_out.map {|e| e['error_name']}.uniq
+    jobs = sql_out.map {|e| e['job_name']}.uniq
+
+    error_score_jobs = {}
+
+    errors.each do |e|
+      jobs.each do |job|
+        flaky_result = run_command('./sql_run.sh calculate_flakiness_jobs.sql', args: [e, FLAKY_BUILDS_DEFAULT_RANGE, job])
+        next if flaky_result.empty?
+        # This is not guaranteed to be 'not consistent', we need to re-check if the last 3 builds were failing because of this
+        flaky_ratio = flaky_result.first['failure_percentage'].to_f/100.0
+
+        job_priority = JOB_PRIORITIES[job]
+        job_priority = job_priority*1.5 if flaky_ratio == 1
+        
+        error_score_jobs[job] = [] if error_score_jobs[job].nil?
+        error_score_jobs[job] << (job_priority*flaky_ratio)
+      end
+    end
+    
+    # Get only maximum score for each job
+    error_score_jobs.each_value.map {|e| e.max}.sum.round(3)
```

Comment:
Why do we need the maximum here? If you are iterating over the error flakiness in each of the jobs, isn't the total priority of them the complete sum?

Reply:
The idea behind this is to reduce the "amount of priority" that we get when an error appears in a lot of jobs.